### ISSUE TITLE
zsh completion throws error v1.10.0-beta.2

### ISSUE THREAD

**Issue by den-is** [kubernetes/kubernetes#61017]
/kind bug

I wasn't able to find issue opened for ZSH case, and aware about similar, fixed, case with bash in 1.10.0-b1 https://github.com/kubernetes/kubernetes/pull/60135
bash completion is working fine

```sh
kubectl create -f calico.y__kubectl_filedir:3: command not found: __debug
__kubectl_filedir:20: command not found: __debug
ml
configmap "calico-config" created
...
```
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.0-beta.2", GitCommit:"63dad40a0391b7af32c34fdbf41fa199c3b247ad", GitTreeState:"clean", BuildDate:"2018-03-07T20:42:18Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.0-beta.2", GitCommit:"63dad40a0391b7af32c34fdbf41fa199c3b247ad", GitTreeState:"clean", BuildDate:"2018-03-07T20:30:35Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```
```
zsh 5.0.2 (x86_64-redhat-linux-gnu)
running latest oh-my-zsh
```

**Comment by tpepper** [comment]
@kubernetes/sig-cli-bugs can you verify this and label for v1.10 milestone and note if it will be considered a release blocker

**Comment by sttts** [comment]
I cannot reproduce this on my local machine (zsh 5.3.1 x86_64-apple-darwin16.3.0).

Please produce a docker container (with Dockerfile) that shows this issue.

**Comment by sttts** [comment]
And another important question for the release: @den-is can you try with the 1.9 kubectl? Is that working in your setup?

**Comment by den-is** [comment]
little bit more info:
i'm running this on CentOS 7.4 1708 x64 and zsh from official repo
same issue happenes with 1.10.0-b3:
```
kubectl version                                                                                                                                                                                    1
Client Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.0-beta.3", GitCommit:"6908cc4291744523ded203e6a5358a2645c43a3d", GitTreeState:"clean", BuildDate:"2018-03-12T21:27:15Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.0-beta.3", GitCommit:"6908cc4291744523ded203e6a5358a2645c43a3d", GitTreeState:"clean", BuildDate:"2018-03-12T21:13:44Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```
1.9.3
```
kubectl create -f cal__kubectl_filedir:3: command not found: __debug                                                                                                                               1
__kubectl_filedir:20: command not found: __debug
ico

kubectl version                                                                                                                                                                                    1
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-07T12:22:21Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.0-beta.3", GitCommit:"6908cc4291744523ded203e6a5358a2645c43a3d", GitTreeState:"clean", BuildDate:"2018-03-12T21:13:44Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by sttts** [comment]
From sig-cli Slack:

Kazuki Suda [01:22]
It seems that `__debug` have to be renamed to `__kubectl_debug` due to changes contained https://github.com/kubernetes/kubernetes/pull/60689.
https://github.com/kubernetes/kubernetes/search?utf8=%E2%9C%93&q=__debug&type= (edited)

**Comment by superbrothers** [comment]
@sttts You can reproduce this with the below:

```
$ minikube start --kubernetes-version=v1.9.0
$ mkdir 61017 && cd 61017
$ cat <<EOL > Dockerfile
FROM k8s.gcr.io/debian-base-amd64:0.3

RUN set -x && \
    apt-get update && \
    apt-get install -y curl zsh && \
    curl -L -o /usr/bin/kubectl https://storage.googleapis.com/kubernetes-release/release/v1.10.0-beta.3/bin/linux/amd64/kubectl && \
    chmod +x /usr/bin/kubectl && \
    echo "autoload -U +X compinit && compinit; source <(kubectl completion zsh)" >> /root/.zshrc
EOL
$ docker build -t 61017 .
$ docker run -it -v ${HOME}/.kube:/root/.kube -v ${HOME}/.minikube:${HOME}/.minikube 61017 zsh
500ff16e30ce# kubectl create -f __kubectl_filedir:3: command not found: __debug
__kubectl_filedir:20: command not found: __debug

bin/    boot/   dev/    etc/    home/   lib/    lib64/  media/  mnt/    opt/    proc/   root/   run/    sbin/   srv/    sys/    tmp/    usr/    var/
500ff16e30ce# kubectl create -f 
```

**Comment by superbrothers** [comment]
`__debug` was renamed to `__kubectl_debug` at https://github.com/spf13/cobra/pull/520/files#diff-3db5cab63dbb32fff4264f5449fd2904R25, but `__debug` has been used yet in completions generated by `kubectl completion zsh`. So we have to replace `__debug` to `__kubectl_debug`.

**Comment by den-is** [comment]
issue still exists with v10.0.0-b4
```
root@kubm1 ~# kubectl create -f __kubectl_filedir:3: command not found: __debug
__kubectl_filedir:20: command not found: __debug
              kubectl create -f kube__kubectl_filedir:3: command not found: __debug
__kubectl_filedir:20: command not found: __debug              kube-dns.yml                   kubelet-api-admin.yml          kube-proxy-rbac.yml
calico.yml    kubectl create -f kube-pro__kubectl_filedir:3: command not found: __debug
__kubectl_filedir:20: command not found: __debug-binding.yml  kubelet-api-admin.yml          kube-proxy-ds.yml              kube-proxy-rbac.yml
                                        xy-__kubectl_filedir:3: command not found: __debug
__kubectl_filedir:20: command not found: __debug
              kubectl create -f kube-proxy-r__kubectl_filedir:3: command not found: __debug
__kubectl_filedir:20: command not found: __debug
bac.yml
serviceaccount "kube-proxy" created
clusterrolebinding.rbac.authorization.k8s.io "system:kube-proxy" created

root@kubm1 ~# kubectl version
Client Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.0-beta.4", GitCommit:"a2dacb6f846e9d64d02c5dc5ab3972287d337405", GitTreeState:"clean", BuildDate:"2018-03-14T05:58:05Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.0-beta.4", GitCommit:"a2dacb6f846e9d64d02c5dc5ab3972287d337405", GitTreeState:"clean", BuildDate:"2018-03-14T05:45:34Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by dixudx** [comment]
Because the fix is not cherry-picked to release-1.10 yet.
